tests/test_goldens.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+import difflib
+import filecmp
+import os
+import shutil
+import sys
+
+from docuploader import shell
+
+import pytest
+
+from sphinx.application import Sphinx
+
+
+@pytest.mark.parametrize(
+    "test_dir", ["handwritten", "gapic-combo", "gapic-auto"]
+)
+def test_goldens(update_goldens, test_dir):
+    source_dir = Path("tests/testdata") / test_dir
+    golden_dir = Path("tests/testdata/goldens") / test_dir
+
+    docs_dir = source_dir / "docs"
+    build_dir = docs_dir / "_build"
+
+    if build_dir.exists():
+        shutil.rmtree(build_dir)
+
+    out_dir = build_dir / "html/docfx_yaml"
+
+    # Install dependencies.
+    try:
+        shell.run(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "install",
+                "-e",
+                ".[all]",
+                "recommonmark",
+            ],
+            cwd=source_dir,
+            hide_output=False,
+        )
+    except Exception as e:
+        pytest.fail(f"build raised an exception: {e}")
+
+    # Generate!
+    try:
+        shell.run(
+            [
+                "sphinx-build",
+                # Settings to be used for sphinx-build
+                "-D",
+                # Extensions to be added
+                ("extensions=sphinx.ext.autodoc,sphinx.ext.autosummary,docfx_yaml.extension," +
```

Comment:
I don't think the + is needed at the end?

Reply:
Done! Did a bit of formatting as well.